src/HttpHandler/Guzzle6HttpHandler.php

```diff
@@ -57,6 +100,41 @@ public function __invoke(RequestInterface $request, array $options = [])
      */
     public function async(RequestInterface $request, array $options = [])
     {
-        return $this->client->sendAsync($request, $options);
+        $requestEvent = null;
+
+        if ($this->logger) {
+            $requestEvent = new LogEvent();
+
+            $requestEvent->method = $request->getMethod();
+            $requestEvent->url = $request->getUri()->__toString();
+            $requestEvent->headers = $request->getHeaders();
+            $requestEvent->payload = $request->getBody()->getContents();
+            $requestEvent->retryAttempt = $options['retryAttempt'] ?? null;
+            $requestEvent->serviceName = $options['serviceName'] ?? null;
+            $requestEvent->clientId = spl_object_id($this->client);
+            $requestEvent->requestId = spl_object_id($request);
```

Comment:
since this code is duplicated above, why not have private methods like `logHttpRequest($request)` and `logHttpResponse($response, $requestEvent)`?

Reply:
In my mind, formatting the `LogEvent` and filling that data is not duty of the `log` function. The `log` function takes the generic log object and turns it into a `json` string to the docs specification, that's why I decided to have this code "duplicated", which in my mind is not really "duplicated" but is basically the same, mostly because one is for the response and one is for the request, and I did it this way in order to keep them separate from the logging logic.

I also thought perhaps a function to make a `logEvent` but voted against it as it would have to be flexible enough to acomodate the small differences between `response` and `request` but it can definitely be done.